crates/commonware-node/src/dkg/manager/actor/mod.rs

```diff
@@ -269,6 +270,21 @@ where
     ) {
         let mut tx = DkgReadWriteTransaction::new(self.db.read_write());
 
+        // Set the sync floor if configured and no state exists. This tells marshal to ignore blocks below this height.
+        if self.config.sync_floor && !tx.has_post_allegretto_state().await {
+            let Ok(floor_height) = self.config.execution_node.provider.best_block_number() else {
+                error!(
+                    "sync_floor enabled but failed to get highest block from execution provider"
+                );
+                return;
+            };
+            info!(
+                floor_height,
+                "sync_floor enabled; setting marshal floor to highest execution block"
+            );
+            self.config.marshal.clone().set_floor(floor_height).await;
```

Comment:
The DKG actor is not the right place for this I think - the floor should already be set when configuring the marshal actor itself.